src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/LanguageServerConstants.cs

```diff
@@ -42,9 +42,7 @@ public static class LanguageServerConstants
 
         public const string RazorResolveCodeActionsEndpoint = "razor/resolveCodeActions";
 
-        public const string RazorProvideSemanticTokensEndpoint = "razor/provideSemanticTokens";
-
-        public const string RazorProvideSemanticTokensEditsEndpoint = "razor/provideSemanticTokensEdits";
+        public const string RazorProvideSemanticTokensRangeEndpoint = "razor/provideSemanticTokensRange";
```

Comment:
@ryanbrandenburg IIRC we removed the range APIs a while back for perf right? I think on startup we'd translate range requests into full document semantic requests which would take years to calculate. Don't know if that'd still be an issue with this but wanted to check

Reply:
We experienced startup perf issues because we received X (where X could be quite large for big files) Range requests on startup, alongside the FullDocument request, which seemed to clog up the UI thread. Yes, our solution was to remove Razors support for it.

On the other hand Roslyn might not experience these issues because they are OOP and surely don't have the same queuing system as us.

I agree that it'd be good to see the perf results from Roslyn though, as I said in my other comment it's suspicious that this didn't help Razor perf.

Reply:
Yep, ideally we'd turn on range handling in Razor to make calls into C# more optimal, but I'm not sure of the Razor implications of that perf- and startup-wise.